File: analysis_moller/hit_analysis_moller/print.py
import ROOT as R

tc1=R.TCanvas()
tc1.Divide(3,2)

tc2=R.TCanvas()
tc2.Divide(3,2)

tc3=R.TCanvas()
tc3.Divide(2,2)

tf1=R.TFile("results.root")

h1=tf1.Get("h_cluster_adc_xplane")
h2=tf1.Get("h_cluster_size_xplane")
h3=tf1.Get("h_cluster_pos_xplane")
h4=tf1.Get("h_strip_adc_xplane")
h5=tf1.Get("h_strip_pos_xplane")

h6=tf1.Get("h_cluster_adc_yplane")
h7=tf1.Get("h_cluster_size_yplane")
h8=tf1.Get("h_cluster_pos_yplane")
h9=tf1.Get("h_strip_adc_yplane")
h10=tf1.Get("h_strip_pos_yplane")

h11=tf1.Get("h_charge_corr")
h12=tf1.Get("h_cluster_2dmap")
h13=tf1.Get("h_strip_timing")

tc1.cd(1)
h1.Draw()
tc1.cd(2)
h2.Draw()
tc1.cd(3)
h3.Draw()
tc1.cd(4)
h4.Draw()
tc1.cd(5)
h5.Draw()

tc2.cd(1)
h6.Draw()
tc2.cd(2)
h7.Draw()
tc2.cd(3)
h8.Draw()
tc2.cd(4)
h9.Draw()
tc2.cd(5)
h10.Draw()

tc3.cd(1)
h11.Draw()
tc3.cd(2)
h12.Draw()
tc3.cd(3)
h13.Draw()


tc1.Draw()
tc2.Draw()
tc3.Draw()

tc1.Print("cosmics.pdf(","pdf")
tc2.Print("cosmics.pdf","pdf")
tc3.Print("cosmics.pdf)","pdf")

File: analysis_moller/hit_analysis_moller/print.cpp
void draw_pad5(TH1F *h1,TH1F *h2,TH1F *h3,TH1F *h4,TH1F *h5,TCanvas *c)
{
    c->cd(1);
    h1->Draw();
    c->cd(2);
    h2->Draw();
    c->cd(3);
    h3->Draw();
    c->cd(4);
    h4->Draw();
    c->cd(5);
    h5->Draw();
    gPad->Update();
}

void draw_pad6(TH2F *h1,TH2F *h2,TH2F *h3,TH2F *h4,TH1F *h5,TH1F *h6,TCanvas *c)
{
    c->cd(1);
    h1->Draw();
    c->cd(2);
    h2->Draw();
    c->cd(3);
    h3->Draw();
    c->cd(4);
    h4->Draw();
    c->cd(5);
    h5->Draw();
    c->cd(6);
    h6->Draw();
    gPad->Update();
}

void print_to_pdf(const char *file1="results.root",int run_no=0)
{
    TFile *f1 = new TFile(file1);

    //setup canvas
    TCanvas *c = new TCanvas("c","Xplane",1900,1000);
    c->Divide(3,2);
    TCanvas *c1 = new TCanvas("c1","YPlane",1900,1000);
    c1->Divide(3,2);
    TCanvas *c2 = new TCanvas("c2","2dPlane",1900,1000);
    c2->Divide(3,2);

    TH1F *h1 = (TH1F*) f1 -> Get("h_cluster_adc_xplane");
    TH1F *h2 = (TH1F*) f1 -> Get("h_cluster_size_xplane");
    TH1F *h3 = (TH1F*) f1 -> Get("h_cluster_pos_xplane");
    TH1F *h4 = (TH1F*) f1 -> Get("h_strip_adc_xplane");
    TH1F *h5 = (TH1F*) f1 -> Get("h_strip_pos_xplane");

    TH1F *h6 = (TH1F*) f1 -> Get("h_cluster_adc_yplane");     
    TH1F *h7 = (TH1F*) f1 -> Get("h_cluster_size_yplane");
    TH1F *h8 = (TH1F*) f1 -> Get("h_cluster_pos_yplane");      
    TH1F *h9 = (TH1F*) f1 -> Get("h_strip_adc_yplane");
    TH1F *h10 = (TH1F*) f1 -> Get("h_strip_pos_yplane");

    TH2F *h11 = (TH2F*) f1 -> Get("h_charge_corr");
    TH2F *h12 = (TH2F*) f1 -> Get("h_charge_corr_s");
    TH2F *h13 = (TH2F*) f1 -> Get("h_charge_corr_c");
    TH2F *h14 = (TH2F*) f1 -> Get("h_cluster_2dmap");
    TH1F *h15 = (TH1F*) f1 -> Get("h_charge_ratio");
    TH1F *h16 = (TH1F*) f1 -> Get("h_strip_timing");

    draw_pad5(h1,h2,h3,h4,h5,c);
    draw_pad5(h6,h7,h8,h9,h10,c1);
    draw_pad6(h11,h12,h13,h14,h15,h16,c2);

    c->Draw();
    c1->Draw();
    c2->Draw();
    
    char *filename = Form("cosmics_%d.pdf(",run_no);
    char *filename1 = Form("cosmics_%d.pdf",run_no);
    char *filename2 = Form("cosmics_%d.pdf)",run_no); 
   
    c->Print(filename);
    c1->Print(filename1);
    c2->Print(filename2);    
}
